pandas/core/frame.py

```diff
@@ -4140,12 +4160,8 @@ def rename(self, *args, **kwargs):
         2  2  5
         4  3  6
         """
-        axes = validate_axis_style_args(self, args, kwargs, "mapper", "rename")
```

Comment:
Side note - there is only one other use now of this `validate_axis_style_args` function (see `DataFrame.reindex`). The intent here is good, but I think the fact that it mutates the args and kwargs being passed it makes it very difficult to reason about. As a follow up to this would like to replace entirely with something simple that just operates on the axis argument

Reply:
Is there anything blocking a similar PR for `.reindex`?

> As a follow up to this would like to replace entirely with something simple that just operates on the axis argument

Is this referring to DataFrame.reindex, or validate_axis_style_args? Ideally we can completely remove `validate_axis_style_args`.